dcos/package.py

```diff
@@ -58,10 +58,13 @@ def install_app(pkg, revision, init_client, options, app_id):
     init_desc = pkg.marathon_json(revision, options)
 
     if app_id is not None:
-        logger.debug('Setting app ID to "%s" (was "%s")',
-                     app_id,
-                     init_desc['id'])
-        init_desc['id'] = app_id
+        if not isinstance(init_desc, list):
+            logger.debug('Setting app ID to "%s" (was "%s")',
+                         app_id,
+                         init_desc['id'])
+            init_desc['id'] = app_id
+        else:
+            logger.debug('Not setting app ID for multi app json')
```

Comment:
There's a number of situations where this would be useful. Mainly it's me not wanting a single overloaded Docker container with a supervisor process inside. Like k8s is currently using.
Think openvpn. You need the openvpn service itself running on the public slaves as well as some administrative service running internally. You could make the user run something like `dcos package install openvpn-gateway; dcos package install openvpn-adminserver` but wouldn't it be nicer if they could just do `dcos package install openvpn`?
Or in my case, I have an edge router consisting of two services. One sets up VRRP on the public slave nodes, the other sets up a haproxy config generator (https://github.com/mesosphere/marathon-lb). I don't want the user to have to run `dcos package install haproxy-vrrp; dcos package install marathon-lb` because the one without the other has no function. Instead installing marathon-lb should start two applications, keepalived which is provided with the dcos-image directly and a marathon-lb docker container. I'd also like to keep the haproxy container unprivileged. Whereas keepalived requires host root privileges to manipulate the network interface IP configuration. If both run in the same privileged container that's a security risk since haproxy is externally accessible.

Removing --app-id should be a separate PR imho.